script/tool/lib/src/update_min_sdk_command.dart

```diff
@@ -69,12 +69,15 @@ class UpdateMinSdkCommand extends PackageLoopingCommand {
         YamlEditor(package.pubspecFile.readAsStringSync());
     if (dartRange != null &&
         (dartRange.min ?? Version.none) < _dartMinVersion) {
+      Version upperBound = _dartMinVersion.nextMajor;
+      // pub special-cases 3.0.0 as an upper bound to be treated as 4.0.0, and
+      // using 3.0.0 is now an error at upload time, so special case it here.
```

Comment:
I had heard about this special casing, I think it only kicks in if your lower bound is modern enough (but I guess packages in this repo have been kept up to date so the 3 -> 4 applies everywhere.)

Reply:
Modern enough is 2.12 (NNBD), so we're definitely way past that.